From c14b65106687b1097f5d50a993707d8a2f99005b Mon Sep 17 00:00:00 2001
From: jiajia Qian <jiajia.qian@nxp.com>
Date: Tue, 19 Dec 2023 13:23:33 +0800
Subject: [PATCH] tests: Fix cl test Include Directories: error : Error(0,8) :
 Cannot find the header file include_test.h.

Modified CMakeLists.txt to install include_test.h file in piglit/tests/cl/program/build/ folder.

Modified the relative path in include-directories.cl from "tests/cl/program/build" to "../lib/piglit/tests/cl/program/build".

Upstream-Status: Pending [https://gitlab.freedesktop.org/mesa/piglit/-/merge_requests/853]

Signed-off-by: jiajia Qian <jiajia.qian@nxp.com>
---
 CMakeLists.txt                                | 1 +
 tests/cl/program/build/include-directories.cl | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dd2bf6712..8b859a3eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -553,6 +553,7 @@ install (
 	DIRECTORY tests
 	DESTINATION ${PIGLIT_INSTALL_LIBDIR}
 	FILES_MATCHING REGEX ".*\\.(xml|xml.gz|py|program_test|shader_test|shader_source|frag|vert|geom|tesc|tese|comp|spv|ktx|cl|txt|inc|vk_shader_test)$"
+	REGEX "include_test.h"
 	REGEX "CMakeFiles|CMakeLists|serializer.py|opengl.py|cl.py|quick_gl.py|glslparser.py|shader.py|quick_shader.py|no_error.py|llvmpipe_gl.py|sanity.py" EXCLUDE
 )
 
diff --git a/tests/cl/program/build/include-directories.cl b/tests/cl/program/build/include-directories.cl
index 40ed9443a..23238f212 100644
--- a/tests/cl/program/build/include-directories.cl
+++ b/tests/cl/program/build/include-directories.cl
@@ -2,7 +2,7 @@
 [config]
 name: Include Directories
 clc_version_min: 10
-build_options: -I . -I tests/cl/program/build
+build_options: -I . -I ../lib/piglit/tests/cl/program/build
 !*/
 
 #include "include_test.h"
-- 
2.34.1

